unique_toolkit/unique_toolkit/language_model/prompt.py

```diff
@@ -0,0 +1,45 @@
+from string import Template
+from typing import Any
+
+from unique_toolkit.language_model.schemas import (
+    LanguageModelSystemMessage,
+    LanguageModelUserMessage,
+)
+
+
+class Prompt:
+    def __init__(self, template: str, **kwargs):
+        self._template = Template(template)
+        self._content = self._template.template
+        if kwargs:
+            self._content = self._template.substitute(**kwargs)
+
+    @property
+    def template(self):
+        return self._template
+
+    @property
+    def content(self):
+        return self._content
+
+    def format(self, **kwargs: Any) -> str:
+        """
+        Formats the template with the given kwargs. Raises KeyError if a required parameter is missing.
+
+        Args:
+            **kwargs: Keyword arguments to substitute into the template.
+
+        Returns:
+            str: The formatted template string.
+
+        Raises:
+            KeyError: If a required parameter in the template is missing from kwargs.
+        """
+        self._content = self._template.substitute(**kwargs)
+        return self._content
+
+    def to_user_msg(self) -> LanguageModelUserMessage:
```

Comment:
Should we raise a warning here if there are unfilled template parameters?

Reply:
An expception would then be thrown when using .substitute. Only .safe_substitute does not throw an exception when parameters have not been filled.

Reply:
Ah nice

Reply:
Mind renaming to substitute then?